tests/conftest.py

```diff
@@ -84,6 +85,9 @@ def call_xfail(self, cmd):
         proc = self.unchecked_call(cmd)
         assert proc.returncode != 0
         assert proc.stderr != ""
+        def f(self2, reg):
+            assert re.search(reg, self2.stderr)
+        proc.match = types.MethodType(f, proc)
```

Comment:
I think you can simplify it a bit like this:
```
def f(reg):
    assert re.search(reg, proc.stderr)
proc.match = f
```

Reply:
I strongly prefer my code as it is, because 1) closures in pythons are difficult and 2) it will become a member function anyway, so it's fine to have a self/this argument.